tests/integration/buildcmd/test_build_in_source.py

```diff
@@ -107,3 +115,63 @@ def test_builds_successfully_with_local_dependency(self):
 
         # check whether dependencies were installed in source dir
         self.assertEqual(os.path.isdir(os.path.join(codeuri, "node_modules")), True)
+
+
+class TestBuildCommand_BuildInSource_Nodejs(BuildIntegNodeBase):
+    is_nested_parent = False
+    template = "template.yaml"
+
+    def setUp(self):
+        super().setUp()
+
+    def tearDown(self):
+        super().tearDown()
+
+    def validate_node_modules(self, is_build_in_source_behaviour: bool):
+        # validate if node modules exist in the built artifact dir
+        built_node_modules = Path(self.default_build_dir, "Function", "node_modules")
+        self.assertEqual(built_node_modules.is_dir(), True, "node_modules not found in artifact dir")
+        self.assertEqual(built_node_modules.is_symlink(), is_build_in_source_behaviour)
+
+        # validate that node modules are suppose to exist in the source dir
+        source_node_modules = Path(self.codeuri_path, "node_modules")
+        self.assertEqual(source_node_modules.is_dir(), is_build_in_source_behaviour)
+
+    @parameterized.expand(
+        [
+            (True, True),  # build in source
+            (False, False),  # don't build in source
+            (None, False),  # use default for workflow (don't build in source)
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_builds_successfully_without_local_dependencies(self, build_in_source, expected_built_in_source):
+        code_uri = "Node"
+        source_path = Path(self.test_data_path, code_uri)
+        self.codeuri_path = Path(self.working_dir, code_uri)
+
+        shutil.copytree(source_path, self.codeuri_path, dirs_exist_ok=True, symlinks=True)
+
+        overrides = self.get_override(
+            runtime="nodejs18.x", code_uri=self.codeuri_path, architecture="x86_64", handler="main.lambdaHandler"
+        )
+        command_list = self.get_command_list(build_in_source=build_in_source, parameter_overrides=overrides, debug=True)
+
+        run_command(command_list, self.working_dir)
+        self.validate_node_modules(expected_built_in_source)
+
+    @pytest.mark.flaky(reruns=3)
+    def test_builds_successfully_with_local_dependency(self):
+        codeuri_folder = Path("Esbuild", "NodeWithLocalDependency")
```

Comment:
Is it possible to restructure the testdata in a way where we can move the copytree logic to the setup methods? The copying logic isn't part of the test case itself but sets up the environment for the tests to run in, I feel like that logic should be abstracted away like in other test classes.

Reply:
Thats a good point. As it is right now, the copy here copies this specific test data folder and not the entire `buildcmd` test data folder to save time. We could move this to the setup and make it more broad (copy the entire folder), or we could just move those two tests into the scratch folder from within setup. I think that restructuring is also possible, but might also need to be updated else where/copied again for this test.

In any case, I can look to improve this.

Reply:
Updated test class setup to copy the `"Esbuild"` folder. We can use this instead for both normal and local dep testing since it contains all the folders for both tests.